hphp/runtime/vm/jit/relocation-arm.cpp

```diff
@@ -571,6 +572,61 @@ size_t relocateImmediate(Env& env, TCA srcAddr, TCA destAddr,
   return true;
 }
 
+/*
+ * Relocates padding. Service request stubs are padded out so that
+ * they are always a fixed size. However, the relocation feature can
+ * request that a large swath of instructions be translated and there can
+ * be multiple service request stubs within. This helper function detects
+ * padding in the source translation and then finds the boundaries of its
+ * service request stub. It then determines how much padding is required
+ * in the output. This may sound like a trivial problem, but on ARM service
+ * requests stubs can grow/shrink due to branch distances. Therfore the
+ * destination may contain a different amount of padding.
+ */
+bool relocatePadding(Env& env, TCA srcAddr, TCA destAddr,
+                     size_t& srcCount, size_t& destCount,
+                     TCA destStart) {
+
+  auto const srcAddrActual = env.srcBlock.toDestAddress(srcAddr);
+  auto src = Instruction::Cast(srcAddrActual);
+
+  if (src->Mask(ExceptionMask) != BRK) return false;
+
+  // Check to see if this is a stub and determine its start.
+  auto srcAddrBegin = (TCA)nullptr;
+  for (auto addr : env.meta.reusedStubs) {
+    if ((srcAddr >= addr) && (srcAddr < (addr + svcreq::stub_size()))) {
+      srcAddrBegin = addr;
+      break;
+    }
+  }
+  if (!srcAddrBegin) return false;
+
+  // Add the src padding to the rewrites.
+  src = src->NextInstruction();
+  env.rewrites.insert(src);
+  while (src->Mask(ExceptionMask) == BRK) {
+    src = src->NextInstruction();
+    env.rewrites.insert(src);
+    srcCount++;
+  }
+
+  // Pad out the remainder of the dest service request stub
+  auto const destAddrBegin = (srcAddrBegin == env.start) ?
+    destStart : env.rel.adjustedAddressAfter(srcAddrBegin);
```

Comment:
question: Is `destStart != env.rel.adjustedAddressAfter(srcAddrBegin)` if `srcAddrBegin == env.start`?